File: Data Structures & Algorithms/duplicate-integer/submission-1.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        if (nums.Length == 0) return false;
        List<int> seen = new List<int> {nums[0]};
        int i = 1;
        while (i < nums.Length)
        {
            if (seen.Contains(nums[i]))
            {
                return true;
            }
            else
            {
                seen.Add(nums[i]);
                i++;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) return false;

        const seen = {};
        for (const i in s) {
            const char = s[i];
            if (seen[char]) seen[char] += 1;
            else seen[char] = 1;
        }
        for (const i in t) {
            const char = t[i];
            if (!seen[char]) return false;
            seen[char]--;
            if (seen[char] === 0) delete seen[char];
        }
        return true;
    }
}
